main.go

```diff
@@ -49,17 +50,134 @@ func makeClient() http.Client {
 	return proxyClient
 }
 
+type NatsQueue struct {
+	clusterID string
+	clientID  string
+	natsURL   string
+
+	maxReconnect   int
+	reconnectDelay time.Duration
+	conn           stan.Conn
+	connMutex      *sync.RWMutex
+	quitCh         chan struct{}
```

Comment:
If `quitCh` is for a graceful shutdown and we're editing the file how about `shutdownCh`?

Reply:
Do we even need a suffix of `ch` in that instance?

Reply:
If there is `ch` suffix it's easier to understand looking into to code. At least for me and I saw that pattern a lot in Go code. If you want I can remove it but I will stand behind the idea to leave suffix.